packages/nodes-base/nodes/RabbitMQ/GenericFunctions.ts

```diff
@@ -1,64 +1,55 @@
-	this: IExecuteFunctions | ITriggerFunctions,
-	options: IDataObject,
-): Promise<amqplib.Channel> {
-	const credentials = await this.getCredentials('rabbitmq');
+const credentialKeys = ['hostname', 'port', 'username', 'password', 'vhost'] as const;
 
-	const credentialKeys = ['hostname', 'port', 'username', 'password', 'vhost'];
-
-	const credentialData: IDataObject = {};
-	credentialKeys.forEach((key) => {
-		credentialData[key] = credentials[key] === '' ? undefined : credentials[key];
-	});
+export async function rabbitmqConnect(
+	credentials: RabbitMQCredentials,
+): Promise<amqplib.Connection> {
+	const credentialData = credentialKeys.reduce((acc, key) => {
+		acc[key] = credentials[key] === '' ? undefined : credentials[key];
+		return acc;
+	}, {} as IDataObject) as amqplib.Options.Connect;
 
 	const optsData: IDataObject = {};
-	if (credentials.ssl === true) {
+	if (credentials.ssl) {
 		credentialData.protocol = 'amqps';
 
 		optsData.ca =
-			credentials.ca === '' ? undefined : [Buffer.from(formatPrivateKey(credentials.ca as string))];
-		if (credentials.passwordless === true) {
+			credentials.ca === '' ? undefined : [Buffer.from(formatPrivateKey(credentials.ca))];
+		if (credentials.passwordless) {
 			optsData.cert =
-				credentials.cert === ''
-					? undefined
-					: Buffer.from(formatPrivateKey(credentials.cert as string));
+				credentials.cert === '' ? undefined : Buffer.from(formatPrivateKey(credentials.cert));
 			optsData.key =
-				credentials.key === ''
-					? undefined
-					: Buffer.from(formatPrivateKey(credentials.key as string));
+				credentials.key === '' ? undefined : Buffer.from(formatPrivateKey(credentials.key));
 			optsData.passphrase = credentials.passphrase === '' ? undefined : credentials.passphrase;
 			optsData.credentials = amqplib.credentials.external();
 		}
 	}
 
+	return await amqplib.connect(credentialData, optsData);
+}
+
+export async function rabbitmqCreateChannel(
+	this: IExecuteFunctions | ITriggerFunctions,
+): Promise<amqplib.Channel> {
+	const credentials = await this.getCredentials<RabbitMQCredentials>('rabbitmq');
+
 	return await new Promise(async (resolve, reject) => {
 		try {
-			const connection = await amqplib.connect(credentialData, optsData);
-
-			connection.on('error', (error: Error) => {
-				reject(error);
-			});
-
-			const channel = (await connection.createChannel().catch(console.warn)) as amqplib.Channel;
-
-			if (
-				options.arguments &&
-				((options.arguments as IDataObject).argument! as IDataObject[]).length
-			) {
-				const additionalArguments: IDataObject = {};
-				((options.arguments as IDataObject).argument as IDataObject[]).forEach(
-					(argument: IDataObject) => {
-						additionalArguments[argument.key as string] = argument.value;
-					},
-				);
-				options.arguments = additionalArguments;
-			}
+			const connection = await rabbitmqConnect(credentials);
+			// TODO: why is this error handler being added here?
+			connection.on('error', reject);
```

Comment:
I still don't know why this line exists, since 
1. the connection can emit an `error` event long after this promise has been resolved
2. the connection isn't going to emit an `error` event when creating a channel. that error is thrown in the `createChannel` call itself.